### PR TITLE
Update Firestore and User Data Classes for Extended Functionality

### PR DESCRIPTION
**Pull request by Nath-Assa**
### Key Changes

- Added `friendsList` field to Firestore database and updated the README to reflect these changes.

- Updated `User` data class:
  - Added `friendsSet` property to manage user connections more effectively.
  - Changed `eventsAttendeeList` and `eventsHostList` from `List` to `MutableSet` to facilitate easier modifications to event participations and hosting details.
 
- Updated `Event` data class:
   - Changed `firebaseID`, `organiserSet`, and `attendeeSet` properties to `val` to enforce immutability, maintaining fixed references while allowing dynamic content updates within the sets.
   - Modified `organiserSet` and `attendeeSet` from  `Set` to `MutableSet` to allow dynamic updates to these collections without changing their references.

### CODE REVIEW

**Comment by Nath-Assa** [comment]
**Considered Changes:**
  - Evaluated the possibility of updating `getEventsByIds` to accept a `MutableSet<String>` instead of a `List<String>` for parameter consistency. Decided against this modification at this time to prioritize more impactful development tasks.

File: app/src/main/java/com/github/se/eventradar/model/event/Event.kt [hunk 1]
```diff
@@ -56,17 +56,17 @@ data class Event(
...
-    map["attendees_list"] = attendeeList.toList()
+    map["organisers_list"] = organiserSet.toList()
+    map["attendees_list"] = attendeeSet.toList()
     map["category"] = category.name
     return map
   }
 }
 
-private fun getSetOfStrings(data: Any?): Set<String> {
+private fun convertToMutableSetOfStrings(data: Any?): MutableSet<String> {
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/model/event/Event.kt:66]: nit: This seems to be reused twice. Would be nice to put this in a reusable place, but not a priority.

**Review by jgeaso1266 — APPROVED** [review]
One nit but LGTM. Please increase coverage if you can.

**Comment by wisdomseeker2001** [comment]
Had a look over it and looks all good to me, nothing to add! 
I think if u can increase coverage would be great but if not focus on new task!

**Review by achillet3 — COMMENTED** [review]
Good consistent changes throughout the project. The coverage on Event.kt is quite low. I think LoginSuccess.kt will help to reproduce the tests that gave coverage for User.kt

**Comment by Nath-Assa** [comment]
Closes #178

**Pull request merged**